src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpAutomaticBraceCompletion.cs

```diff
@@ -553,5 +553,26 @@ class C
 }",
 assertCaretPosition: true);
         }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        [Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        [WorkItem(18104, "https://github.com/dotnet/roslyn/issues/18104")]
+        public void CompleteStatementTriggersCompletion()
+        {
+            SetUpEditor(@"
+class Program
+{
+    static void Main(string[] args)
+    {
+        Main$$
+    }
+}");
+
+            VisualStudio.Editor.SendKeys("(ar");
+            VisualStudio.Editor.Verify.CurrentLineText("Main(ar$$)", assertCaretPosition: true);
+
+            VisualStudio.Editor.SendKeys(";");
+            VisualStudio.Editor.Verify.CurrentLineText("Main(args);$$", assertCaretPosition: true);
```

Comment:
1. great.  thanks for this test. having at least one integration tests is *definitely* what we want.
2. that said, this is a complex feature with lots of cases, and lots of potential 'mess up' points with things like formatting/brace-matching.  To me, the interesting tests are the one that validate this behavior with all/several of them active.  I can accept 'brace completion' maybe not working in the tests and having that tracked with https://github.com/dotnet/roslyn/issues/31045, however we really should have *unit* tests that exercise the formatting/normal-completion codepaths without *integration* tests.

Note: a primary concern here is just stability. Our integration tests are awful, often fail, and often go unnoticed.  Whereas, say, the CSharpCompletionCommandHandlerTests have been running successfully for years and cover *huge* swaths of interesting interaction cases.  I'd like us to have the same here.  Thanks!

Reply:
Improved now. This is a rather large test hole across other features so the majority of this work is deferred to #31045.